### DISCUSSION TITLE
Problem with culling or alpha blending ?

### DISCUSSION THREAD

**Discussion by Mrt-Grdjn** [bevyengine/bevy#8181] [Q&A]
Hello,

I'm trying to make a voxel game so I create a mesh and pass it to bevy, the problem is that for some reason when I'm bellow the top face of a voxel, i can see through it :
(vertex color is [0.0, 1.0, 0.0, 1.0] so opaque and alpha blending is set to Blend)
![blend](https://user-images.githubusercontent.com/68290282/227309277-783eeb67-1562-4752-a36e-c642117e0e3f.png)
But i have a top face :
![top](https://user-images.githubusercontent.com/68290282/227308908-d8c026cc-64fa-43a0-a8a7-5dce918d6a1a.png)

I've already checked in wireframe mode and all my vertices/triangles are there so it's not that i'm missing a face. I've also tried no backface culling, all the alpha blend mode and double sided and none of them fixes the issue.

Mesh asset + material :
```rust

let mut chunk_mesh = Mesh::new(mesh_data.primitive_type);
chunk_mesh.insert_attribute(Mesh::ATTRIBUTE_POSITION, mesh_data.vert_pos);
chunk_mesh.insert_attribute(Mesh::ATTRIBUTE_NORMAL, mesh_data.normals);
chunk_mesh.set_indices(Some(Indices::U32(mesh_data.triangle_index)));
chunk_mesh.insert_attribute(Mesh::ATTRIBUTE_COLOR, mesh_data.colors);

let mesh_handle = meshes.add(chunk_mesh);

let mut material: StandardMaterial = Color::rgba(1., 1., 1., 1.).into();
material.alpha_mode = AlphaMode::Blend;

commands.spawn(PbrBundle {
       mesh: mesh_handle,
       material: materials.add(material),
       ..Default::default()
});
```

I don't really know why it happens since the gpu should do a depth test to see which vertex/face is the closest to the camera and clip the one behind.

If you have any idea of how to fix this issue it would be appriciated.

Here are some info about my system, all driver up to date :
```
2023-03-23T18:11:19.011327Z  INFO bevy_winit::system: Creating new window "Bevy App" (0v0)
2023-03-23T18:11:19.146319Z  INFO bevy_render::renderer: AdapterInfo { name: "AMD Radeon RX 6900 XT", vendor: 4098, device: 29615, device_type: DiscreteGpu, driver: "AMD proprietary driver", driver_info: "23.3.1 (AMD proprietary shader compiler)", backend: Vulkan }
2023-03-23T18:11:19.453422Z  INFO bevy_diagnostic::system_information_diagnostics_plugin::internal: SystemInfo { os: "Windows 10 Pro", kernel: "19045", cpu: "AMD Ryzen 9 5900X 12-Core Processor", core_count: "12", memory: "31.9 GiB" }
```

**Comment by tim-blackbird** [comment]
Setting the alpha mode to `Blend` disables depth writes because that would cause other issues:
> Unfortunately, although the depth buffer allows opaque objects to be drawn in arbitrary order cor-
rectly, it has issues with transparent objects. The first issue, is that alpha blended fragments, even if
they have an alpha value of 0, result in a write to the depth-buffer. So even a completely zero-alpha
object was rendered to the screen, it would affect the depth-buffer, with potentially adverse affects.
Writing to the depth buffer could be temporarily disabled to avoid this, but then a partially-alpha
blended object could be overdrawn by an object further away, due to the lack of depth information.

Excerpt from page 3 of [this pdf](https://research.ncl.ac.uk/game/mastersdegree/graphicsforgames/transparencyanddepth/Tutorial%204%20-%20Transparency%20and%20Depth.pdf)

If you need some of your voxels to have transparency they would need to be on a separate entity to avoid both issues.

**Reply by Mrt-Grdjn** [reply]
Oh ok, thank you for the information !
